core/Types.h

```diff
@@ -231,8 +222,7 @@ template <> inline bool isa_type<GroundType>(const TypePtr &what) {
     }
 }
 
-class ProxyType;
-template <> inline bool isa_type<ProxyType>(const TypePtr &what) {
+inline bool is_proxy_type(const TypePtr &what) {
```

Comment:
Can we mention all the types in the `switch` below? I want people to be forced to consider whether they are a proxy type or a ground type if they add a new `TypePtr::Tag`.

Reply:
I believe it used to be that exactly one of `is_ground_type` or `is_proxy_type` would be true. Can we implement one of these two helpers in terms of the other so that it remains true?

Reply:
Oh, sure!

Reply:
@jez The following are neither ground nor proxy:

* LambdaParam
* SelfTypeParam
* AliasType
* SelfType
* TypeVar
* AppliedType

Reply:
Done; I made them explicit lists w/ no default.